js/builtins/weakrefs/cleanup-prevented-with-reference.optional.any.js

```diff
@@ -0,0 +1,56 @@
+// META script=resources/maybe-garbage-collect.js
+// ├──> maybeGarbageCollectAndCleanupAsync
+// └──> resolveGarbageCollection
+/*---
+esid: sec-finalization-registry.prototype.cleanupSome
+info: |
+  FinalizationRegistry.prototype.cleanupSome ( [ callback ] )
+
+  1. Let finalizationRegistry be the this value.
+  2. If Type(finalizationRegistry) is not Object, throw a TypeError exception.
+  3. If finalizationRegistry does not have a [[Cells]] internal slot, throw a TypeError exception.
+  4. If callback is not undefined and IsCallable(callback) is false, throw a TypeError exception.
+  5. Perform ? CleanupFinalizationRegistry(finalizationRegistry, callback).
+  6. Return undefined.
+---*/
+
+let holdingsList = [];
+function cb(holding) {
+  holdingsList.push(holding);
+};
+let finalizationRegistry = new FinalizationRegistry(function() {});
+
+let referenced = {};
+
+function emptyCells() {
+  let target = {};
+  finalizationRegistry.register(target, 'target!');
+  finalizationRegistry.register(referenced, 'referenced');
+
+  let prom = maybeGarbageCollectAndCleanupAsync(target);
+  target = null;
+
+  return prom;
+}
+
+async function execAsync() {
+  try {
+    await emptyCells();
+    finalizationRegistry.cleanupSome(cb);
+
+    assert_equals(holdingsList.length, 1);
+    assert_equals(holdingsList[0], 'target!');
+    assert_equals(typeof referenced, 'object', 'referenced preserved');
+  } catch (exception) {
+    return exception;
+  }
+}
+
+promise_test(() => {
+  assert_implements(
```

Comment:
@jugglinmike I've intentionally used `assert_implements` because the function we need is not optional, so these should fail when `cleanupSome` is not present.